src/main/java/com/bookshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.bookshop.service.impl;
+
+import com.bookshop.dto.cart.ShoppingCartResponseDto;
+import com.bookshop.dto.cartitem.CartItemResponseDto;
+import com.bookshop.dto.cartitem.CreateCartItemDto;
+import com.bookshop.dto.cartitem.UpdateCartItemDto;
+import com.bookshop.exception.EntityNotFoundException;
+import com.bookshop.mapper.CartItemMapper;
+import com.bookshop.mapper.ShoppingCartMapper;
+import com.bookshop.model.CartItem;
+import com.bookshop.model.ShoppingCart;
+import com.bookshop.model.User;
+import com.bookshop.repository.CartItemRepository;
+import com.bookshop.repository.ShoppingCartRepository;
+import com.bookshop.repository.UserRepository;
+import com.bookshop.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
```

Comment:
I don't know if you need @Transactional on each method. I need that only on update method. Try to test it without transactional, and put it only where you got exception.